### PR TITLE
feat(fe): redesign contest table

### PR DESCRIPTION
**Pull request by youznn**
### Description

client contest table을 리디자인 했습니다.
- search bar 추가
- 로그인 했을 경우 registered 탭 추가
- table column 변경
image
로그인했을 경우 
image
로그인 안 했을 경우

### Additional context

추후 score, registered 여부 추가
status badge 변경 예정

---

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md)
- [x] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md#pr-and-branch) and follow the [Commit Convention](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md#commit-convention)
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: apps/frontend/app/(main)/contest/_components/TableSwitchButton.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+'use client'
+
+import { cn } from '@/lib/utils'
+import Link from 'next/link'
+import { useSearchParams } from 'next/navigation'
+
+export default function TableSwitchButton() {
+  const searchParam = useSearchParams()
```
Comment (B0XERCAT) [inline apps/frontend/app/(main)/contest/_components/TableSwitchButton.tsx:8]:
```
export default function TableSwitchButton({
  searchParams
}: {
  searchParams: { registered?: boolean }
}) {
```
useSearchParams 대신에 위와 같이 사용하면 'use clint'도 뗄 수 있을 것 같아요

File: apps/frontend/app/(main)/contest/_components/RegisteredTableColumns.tsx [hunk 2]
```diff
@@ -19,28 +19,27 @@ export const columns: ColumnDef<Contest>[] = [
...
     accessorKey: 'status',
     cell: ({ row }) => (
-      <Badge type={row.original.status}>
-        <p>
-          {row.original.status.startsWith('registered')
-            ? 'registered'
-            : row.original.status}
-        </p>
-      </Badge>
+      <>
```
Comment (B0XERCAT) [inline apps/frontend/app/(main)/contest/_components/RegisteredTableColumns.tsx:22]: <>는 불필요한 태그 같아요!

File: apps/frontend/app/(main)/contest/_components/FinishedContestTable.tsx [hunk 3]
```diff
@@ -18,20 +30,17 @@ export default async function FinishedContestTable() {
 
   return (
     <>
```
Comment (B0XERCAT) [inline apps/frontend/app/(main)/contest/_components/FinishedContestTable.tsx:32]: 불필요한 태그인 것 같습니다!

File: apps/frontend/app/(main)/contest/_components/TableSwitchButton.tsx (hunk 1 above)
Comment (youznn) [reply apps/frontend/app/(main)/contest/_components/TableSwitchButton.tsx:8]: 오 좋아용

**Review by B0XERCAT — APPROVED** [review]
고생하셨습니다!

**Pull request merged**